lib/swagger/swagger.ex

```diff
@@ -2,127 +2,122 @@ defmodule Xcribe.Swagger do
   @moduledoc """
   Treats list of Requests and generates OpenAPI 3.0 JSON
   """
+
   alias Xcribe.Config
   alias Xcribe.Swagger.{Descriptor, Formatter}
 
+  @empty_data [nil, %{}, []]
+  @security_scheme %{
+    "securitySchemes" => %{
+      "api_key" => %{"name" => "Authorization", "type" => "apiKey", "in" => "header"}
+    }
+  }
+
   def generate_doc(requests) do
-    swagger_json()
-    |> add_requests(requests)
-    |> add_security(requests)
+    requests
+    |> Enum.sort_by(& &1.status_code)
+    |> build_openapi_object()
+    |> include_security_scheme(requests)
     |> Xcribe.JSON.encode!()
   end
 
-  defp swagger_json() do
+  defp build_openapi_object(requests) do
     %{
       "openapi" => "3.0.0",
       "info" => %{
         "title" => Map.get(xcribe_info(), :name, ""),
         "version" => Map.get(xcribe_info(), :version, "0.1.0"),
         "description" => Map.get(xcribe_info(), :description, "")
-      }
+      },
+      "paths" => paths_from_requests(requests)
     }
   end
 
-  defp add_requests(swagger_map, requests) do
-    paths =
-      requests
-      |> Enum.sort_by(& &1.status_code)
-      |> Enum.reduce(%{}, fn x, acc ->
-        Map.put(acc, x.path, Map.merge(acc[x.path] || %{}, handle_request(x, acc)))
-      end)
+  defp paths_from_requests(requests),
+    do: Enum.reduce(requests, %{}, &include_request_into_paths/2)
 
-    Map.put(swagger_map, "paths", paths)
+  defp include_request_into_paths(%{path: path} = request, paths) do
```

Comment:
instead create two function:

- `include_request_into_paths` 
- `include_request_into_given_path` 

you can use `pattern match` to make this decision for you

```elixir
defp include_request(%{path: path} = request, paths) do
  # ...
end

defp include_request(path, %{verb: verb} = request) do
  # ...
end

Reply:
I usually prefer pattern matching as you suggested. But in this case the functions behaviour is very different. So I tried give a suggestive name for each. Maybe the current names are not so good. wdyt?

Reply:
Got it. 

So the two functions are different proposes?
If yes, I agree to set a different name.

My problem is with `into`, that word breaks the reading.

WDYT 

- include_request_by_paths
- include_request_by_given_path

It's almost what you put, but, maybe the reading sounds better.

If you prefer `into` it's ok for me too. =D

Reply:
You have a point. The names was bad. I improved the as:

include_request_into_paths           ->  build_path_from_request
include_request_into_given_path  -> build_path_verb_from_request
include_request_into_given_verb  -> build_responses_from_request